code/modules/surgery/organs/stomach.dm

```diff
@@ -19,34 +19,40 @@
 	var/disgust_metabolism = 1
 
 /obj/item/organ/stomach/on_life()
-	var/mob/living/carbon/human/H = owner
-	var/datum/reagent/Nutri
-
 	..()
-	if(istype(H))
+
+	//Manage species digestion
+	if(istype(owner, /mob/living/carbon/human))
+		var/mob/living/carbon/human/humi = owner
 		if(!(organ_flags & ORGAN_FAILING))
-			H.dna.species.handle_digestion(H)
-		handle_disgust(H)
+			humi.dna.species.handle_digestion(humi)
+
+	//digest food
+	var/mob/living/carbon/body = owner
+	reagents.metabolize(body, can_overdose=TRUE)
+	handle_disgust(body)
 
 	if(damage < low_threshold)
 		return
 
-	Nutri = locate(/datum/reagent/consumable/nutriment) in H.reagents.reagent_list
+	var/datum/reagent/Nutri = locate(/datum/reagent/consumable/nutriment) in reagents.reagent_list
 
 	if(Nutri)
 		if(prob((damage/40) * Nutri.volume * Nutri.volume))
-			H.vomit(damage)
-			to_chat(H, "<span class='warning'>Your stomach reels in pain as you're incapable of holding down all that food!</span>")
+			body.vomit(damage)
+			to_chat(body, "<span class='warning'>Your stomach reels in pain as you're incapable of holding down all that food!</span>")
 
 	else if(Nutri && damage > high_threshold)
 		if(prob((damage/10) * Nutri.volume * Nutri.volume))
-			H.vomit(damage)
-			to_chat(H, "<span class='warning'>Your stomach reels in pain as you're incapable of holding down all that food!</span>")
+			body.vomit(damage)
+			to_chat(body, "<span class='warning'>Your stomach reels in pain as you're incapable of holding down all that food!</span>")
```

Comment:
1: Nutri brings me great pain, please ritually de-soulify it to obey snake_case
2: Am I already losing my mind, or will that else if check never fire.

Reply:
the second one indeed wont fire

Reply:
This is a mess i agree, I will try to clean it up.

Reply:
The new version should now be much better.